services/cloudrun-service/pkg/cloudrun/cloudrun.go

```diff
@@ -0,0 +1,186 @@
+		serviceCallResp, err = serviceCall.Do()
+		if err != nil {
+			return err
+		}
+	} else {
+		serviceCall := runService.Projects.Locations.Services.Create(parent, svc)
+		serviceCallResp, err = serviceCall.Do()
+		if err != nil {
+			return err
+		}
+	}
+	if err := waitForOperation(parent, serviceCallResp, 60); err != nil {
+		return err
+	}
+	getServiceCall := runService.Projects.Locations.Services.Get(servicePath)
+	serviceResp, err := getServiceCall.Do()
+	if err != nil {
+		return err
+	}
+	condition, err := GetServiceReadyCondition(serviceResp)
+	if err != nil {
+		return err
+	}
+	if condition.Status != "True" {
+		return fmt.Errorf("service not ready: %s", condition)
+	}
+	return nil
+}
+
+func GetServiceReadyCondition(serviceCallResponse *run.Service) (ServiceReadyCondition, error) {
+	//exhaustruct:ignore
+	serviceReadyCondition := ServiceReadyCondition{
+		Status:   "",
+		Name:     serviceCallResponse.Metadata.Name,
+		Revision: serviceCallResponse.Status.ObservedGeneration,
+	}
+	conditions := serviceCallResponse.Status.Conditions
+	for _, condition := range conditions {
+		if condition.Type == serviceReady {
+			serviceReadyCondition.Status = condition.Status
+			serviceReadyCondition.Reason = condition.Reason
+			serviceReadyCondition.Message = condition.Message
+		}
+	}
+	if serviceReadyCondition.Status == "" {
+		return serviceReadyCondition, fmt.Errorf("failed to get Ready status for service %s", serviceReadyCondition.Name)
+	}
+	return serviceReadyCondition, nil
+}
+
+func getOperationId(parent string, serviceCallResp *run.Service) (string, error) {
+	operationId, exists := serviceCallResp.Metadata.Annotations[operationIdAnnotation]
+	if !exists {
+		return "", fmt.Errorf("failed to get operation-id for service %s", serviceCallResp.Metadata.Name)
+	}
+	return fmt.Sprintf("%s/operations/%s", parent, operationId), nil
+}
+
+func waitForOperation(parent string, serviceCallResp *run.Service, timeoutSeconds uint16) error {
+	operationId, err := getOperationId(parent, serviceCallResp)
+	if err != nil {
+		return err
+	}
+	opService := run.NewProjectsLocationsOperationsService(runService)
+	//exhaustruct:ignore
+	waitOperationRequest := &run.GoogleLongrunningWaitOperationRequest{
+		Timeout: fmt.Sprintf("%ds", timeoutSeconds),
+	}
+	opServiceCall := opService.Wait(operationId, waitOperationRequest)
+	operationResp, err := opServiceCall.Do()
+	if err != nil {
+		return fmt.Errorf("failed to wait for the service %s: %s", serviceCallResp.Metadata.Name, err)
+	}
+	if !operationResp.Done {
+		return fmt.Errorf("service %s creation exceeded the timeout of %d seconds", serviceCallResp.Metadata.Name, timeoutSeconds)
+	}
+	return nil
+}
+func isPresent(services []*run.Service, serviceName string) bool {
+	for _, service := range services {
+		if service.Metadata.Name == serviceName {
+			return true
+		}
+	}
+	return false
+}
+
+func getParent(svc *run.Service) (string, error) {
```

Comment:
is `getParent` the right name? Maybe `getParentLocation`?

Reply:
It is defined in the [API](https://pkg.go.dev/google.golang.org/api/run/v1#ProjectsLocationsServicesService.Create) as parent.